### ISSUE TITLE
i18n

### ISSUE THREAD

**Issue by andreypopp** [prometheusresearch/react-ui#2]
i18n should be designed along with component library not as the afterthought.

* `paddingStart` and `paddingEnd` instead of `paddingLeft` and `paddingRight` (same for margins, ...)
* minimal contractual API for configuring i18n (React context, global)

cc @jayclassless

**Comment by jayclassless** [comment]
Agreed. Although I would suggest "in addition to" rather than "instead of", just in case the author absolutely needs to do things in a right/left fashion.

Properties that should be addressed:

- `marginLeft`/`marginRight`
- `paddingLeft`/`paddingRight`
- `left`/`right`
- `textAlign`
- The "attach" variants of Button

Would we want to get super fancy and try to parse the values of `margin` and `padding` so that left/right values are swapped?

**Comment by andreypopp** [comment]
> Would we want to get super fancy and try to parse the values of margin and padding so that left/right values are swapped?

I think it's better not to support shortcut forms at all. That means using `margin` is only possible if you want all sides have the same margin: `margin={10}` is ok but `margin="10 20"` is disallowed.

**Comment by jayclassless** [comment]
> I think it's better not to support shortcut forms at all. That means using margin is only possible if you want all sides have the same margin: margin={10} is ok but margin="10 20" is disallowed.

I think that's fair.

**Comment by jayclassless** [comment]
Should also address the `icon`/`iconRight` props on buttons.
